pyBumpHunter/BumpHunter.py

```diff
@@ -286,415 +336,433 @@ def Reset(self):
-        
+
         Arguments :
             data : Numpy array containing the data distribution. This distribution will be transformed into a binned
                    histogram and the algorithm will look for the most significant excess.
-            
+
             bkg : Numpy array containing the background reference distribution. This distribution will be transformed
                   into a binned histogram and the algorithm will compare it to data while looking for a bump.
-            
+
             is_hist : Boolean that specify if the given data and background are already in histogram form. If true,
                       data and bkg are histograms and the Bins argument is expected to be a array with the bound of
                       the bins and the Weights argument is expected to be a scalar scale factor or None.
                       Default to False.
-            
+
             do_pseudo : Boolean specifying if pesudo data should be generated.
                         If False, then the BumpHunter statistics distribution kept in memmory is used to compute
                         the global p-value and significance. If there is nothing in memmory, the global p-value
                         and significance will not be computed.
                         Default to True.
-        
+
         Result inner variables :
             global_Pval : Global p-value obtained from the test statistic distribution.
-            
+
             res_ar : Array of containers containing all the p-value calculated durring the scan of the data (indice=0)
                      and of the pseudo-data (indice>0). For more detail about how the p-values are sorted in the
                      containers, please reffer the the doc of the function scan_hist.
-            
+
             min_Pval_ar : Array containing the minimum p-values obtained for the data (indice=0) and and the pseudo-
                           data (indice>0).
-            
+
             min_loc_ar : Array containing the positions of the windows for which the minimum p-value has been found
                          for the data (indice=0) and pseudo-data (indice>0).
-            
+
             min_width_ar : Array containing the width of the windows for which the minimum p-value has been found for
                            the data (indice=0) and pseudo-data (indice>0).
-            
+
             signal_eval : Number of signal events evaluated form the last scan.
-        '''
-        
+        """
+
         # Set the seed if required (or reset it if None)
         np.random.seed(self.seed)
-        
+
         # Generate the background and data histograms
-        print('Generating histograms')
-        if(is_hist is False):
-            bkg_hist,Hbin = np.histogram(bkg,bins=self.bins,weights=self.weights,range=self.rang)
-            data_hist = np.histogram(data,bins=self.bins,range=self.rang)[0]
+        print("Generating histograms")
```

Comment:
All this prints... I would suggest to add a `verbose` flag or similar, this statements can be annoying. E.g. if someone runs toys and executes it 100- 1000 times. People will have use-cases that you never even thought about ;)

Reply:
You can use the logging module, then a user can activate the logging (or quiet it) if they want.